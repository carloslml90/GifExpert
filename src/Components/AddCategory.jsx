import { useState } from "react";
import Proptypes from 'prop-types';



export const AddCategory=({onNewCategory})=>{

    const [inputValue,setInputValue]=useState(' ')
    //console.log(inputValue);


        const onInputChange=(event) =>{
        setInputValue(event.target.value)
        }

        const onSubmit =(event)=>{
            event.preventDefault();
            if(inputValue.trim().length <=4 ) return;
            onNewCategory(inputValue.trim());
            setInputValue("");//clear our input box after added an element to categories 
  
        }

                return(
                
                
                    <form  onSubmit={(event)=>onSubmit(event)} aria-label="form">

                        <input
                        type="text"
                        value={inputValue}
                        placeholder="Looking for a serie"
                        onChange={(event)=>onInputChange(event)}
                        
                        />
                    </form>
                
                )

} 

AddCategory.propTypes= {

   onNewCategory: Proptypes.func,

}