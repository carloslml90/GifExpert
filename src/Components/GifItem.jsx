import PropTypes from "prop-types";


export const GifItem=({id, title, url})=>{

    console.log({id, title, url})
return(


<div className="card">

<img src={url} alt={title} />
<p id="title-img">{title}</p>

</div>

)





}

GifItem.propTypes={
id:PropTypes.string.isRequired,
title: PropTypes.string.isRequired,
url:PropTypes.string,

}