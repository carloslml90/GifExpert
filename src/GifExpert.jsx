import { useState } from "react"
import { AddCategory, GifGrid } from "./Components"






export const GifExpert=()=>{

const [categories,setCategories]=useState(['Breaking bad']);


//This function write in my input a new category to submit
        const onAddCategory=(newCategory)=>{
          if (categories.includes(newCategory)) return;
        setCategories([newCategory,...categories])
        } 



console.log (categories);

        return(
             <>

                 {/*title */}
                <h1>Gifs Expert </h1>

                  {/*Input */}

                  <AddCategory
                  
                  //setCategories={setCategories}
                  
                  onNewCategory={(event)=>onAddCategory(event)}
                  />

           
              
                    {
                    

                      categories.map((category) =>(


                        <GifGrid key={category}
                        category={category}
                     
                        />
                    ))
                    
                    }

                    
                  


                </>
        ) 

}