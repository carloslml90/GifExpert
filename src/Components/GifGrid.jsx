
import { GifItem } from "./GifItem";

import Proptypes from "prop-types"
import { useFetchGifs } from "../Hooks/useFetchHook";




export const GifGrid=({category})=>{

const {images, isLoading }=useFetchGifs(category);
console.log({isLoading});





    return(
        <>

 
    <h3>{category}</h3>

    {
    isLoading && (<h2>cargando...</h2>)
    }

       <div className="card-grid">

       
            {
            
            images.map(( image ) =>(

                <GifItem  
                key={image.id}
                {...image}
                />
              ))
            }
      </div>
    </>

    )


} 

GifGrid.propTypes={
category: Proptypes.string.isRequired,

}