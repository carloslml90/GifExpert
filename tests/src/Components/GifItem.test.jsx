import { render,screen} from "@testing-library/react";
import { GifItem } from "../../../src/Components";

describe('To prove our test suite',()=>{
    
    const title="Improving every day";
    const url="https://www.saitama.com/saitama.jpg";
    
    test( 'Making matching with snapshot', ()=>{

        const{container}=render(<GifItem  title={title} url={url}/>);
        
        expect(container).toMatchSnapshot();
        
        
    })

    test('Should show the  image with a correct url',()=>{

        render(<GifItem  title={title} url={url}/>);
       /*   screen.debug();

         expect(screen.getByRole('img').src).toBe(url);
         expect(screen.getByRole('img').alt).toBe(title); */

         const {src, alt}=screen.getByRole('img');
         expect(src).toBe(url);
         expect(alt).toBe(alt);
        })
        
        
    test('Should show my image title',()=>{
        
        render(<GifItem  title={title} url={url}/>);

        expect(screen.getAllByText(title)).toBeTruthy();

        
    })     
    



})


