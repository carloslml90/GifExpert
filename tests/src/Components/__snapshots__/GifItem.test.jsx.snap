// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`To prove our test suite Making matching with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Improving every day"
      src="https://www.saitama.com/saitama.jpg"
    />
    <p>
      Improving every day
    </p>
  </div>
</div>
`;
